Microsoft.Toolkit.Uwp.UI/Extensions/ScrollViewer/ScrollViewerExtensions.MiddleClickScrolling.cs

```diff
@@ -0,0 +1,383 @@
+        {
+            if (d is ScrollViewer scrollViewer)
+            {
+                _scrollViewer = scrollViewer;
+            }
+            else
+            {
+                _scrollViewer = (d as FrameworkElement).FindDescendant<ScrollViewer>();
+
+                if (_scrollViewer == null)
+                {
+                    (d as FrameworkElement).Loaded += (sender, arg) =>
+                    {
+                        _scrollViewer = (sender as FrameworkElement).FindDescendant<ScrollViewer>();
+
+                        if (_scrollViewer != null)
+                        {
+                            UpdateChange((bool)e.NewValue);
+                        }
+                    };
+                }
+            }
+
+            if (_scrollViewer == null)
+            {
+                return;
+            }
+
+            UpdateChange((bool)e.NewValue);
+        }
+
+        /// <summary>
+        /// Function to update changes in <see cref="EnableMiddleClickScrollingProperty"/>
+        /// </summary>
+        /// <param name="newValue">New value from the <see cref="EnableMiddleClickScrollingProperty"/></param>
+        private static void UpdateChange(bool newValue)
+        {
+            if (newValue)
+            {
+                _scrollViewer.PointerPressed -= ScrollViewer_PointerPressed;
+                _scrollViewer.PointerPressed += ScrollViewer_PointerPressed;
+            }
+            else
+            {
+                _scrollViewer.PointerPressed -= ScrollViewer_PointerPressed;
+                UnsubscribeMiddleClickScrolling();
+            }
+        }
+
+        /// <summary>
+        /// Function to set default value and subscribe to events
+        /// </summary>
+        private static void SubscribeMiddleClickScrolling()
+        {
+            _isPressed = true;
+            _isMoved = false;
+            _startPosition = default(Point);
+            _currentPosition = default(Point);
+            _isDeferredMovingStarted = false;
+            _oldCursorID = 100;
+            _timer = new Timer(Scroll, null, 5, 5);
+            _isCursorAvailable = IsCursorResourceAvailable();
+
+            Window.Current.CoreWindow.PointerMoved -= CoreWindow_PointerMoved;
+            Window.Current.CoreWindow.PointerReleased -= CoreWindow_PointerReleased;
+
+            Window.Current.CoreWindow.PointerMoved += CoreWindow_PointerMoved;
+            Window.Current.CoreWindow.PointerReleased += CoreWindow_PointerReleased;
+        }
+
+        /// <summary>
+        /// Function to set default value and unsubscribe to events
+        /// </summary>
+        private static void UnsubscribeMiddleClickScrolling()
+        {
+            _isPressed = false;
+            _isMoved = false;
+            _startPosition = default(Point);
+            _currentPosition = default(Point);
+            _isDeferredMovingStarted = false;
+            _oldCursorID = 100;
+            _timer.Dispose();
```

Comment:
_timer could be null here if SubscribeMiddleClickScrolling is not called before

Reply:
Done